tidy3d/web/api/autograd/autograd.py

```diff
@@ -58,7 +58,7 @@ def is_valid_for_autograd(simulation: td.Simulation) -> bool:
         return False
 
     # if too many structures, raise an error
-    structure_indices = [i for key, i, *_ in traced_fields.keys() if key == "structures"]
+    structure_indices = {i for key, i, *_ in traced_fields.keys() if key == "structures"}
```

Comment:
Probably I don't understand this - so indices will always refer to the same object, even if the index appears multiple times?

Reply:
yea, so the field paths can look something like
```
('structures', 0, 'medium', 'permittivity')
('structures', 0, 'geometry', 'size', 1)
('structures', 1, 'geometry', 'vertices')
('structures', 1, 'geometry', 'geometries', 3, 'center', 0)

```

we have a limit only on the number of unique structures, not on the number of traced fields. So now I just count every structure that has traced fields in this check. Hence the set instead of list.